hack/istio/install-topology-generator-demo.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+##############################################################################
+# install-scale-mesh-demo.sh
+#
+# Installs the kiali topology-generator demo application
+# https://github.com/kiali/demos/tree/master/topology-generator
+# Works on both openshift and non-openshift environments.
+##############################################################################
+
+: ${CLIENT_EXE:=oc}
+: ${DELETE_DEMOS:=false}
+: ${DELETE_CONFIG:=false}
+: ${TOPO:=topology-generator}
+: ${BASE_URL:=https://raw.githubusercontent.com/kiali/demos/master}
+: ${NUM_NS:=1}
+
+apply_network_attachment() {
+  NAME=$1
+cat <<NAD | $CLIENT_EXE -n ${NAME} apply -f -
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: istio-cni
+  labels:
+   generated-by: mimik
+NAD
+    cat <<SCC | $CLIENT_EXE apply -f -
+apiVersion: security.openshift.io/v1
+kind: SecurityContextConstraints
+metadata:
+  name: ${NAME}-scc
+runAsUser:
+  type: RunAsAny
+seLinuxContext:
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+users:
+- "system:serviceaccount:${NAME}:default"
+- "system:serviceaccount:${NAME}:${NAME}"
+SCC
+}
+
+install_topology_generator_demo() {
+
+    if [ "${IS_OPENSHIFT}" == "true" ]; then
+        ${CLIENT_EXE} new-project ${TOPO}
+      else
+        ${CLIENT_EXE} create ns ${TOPO}
+    fi
+    ${CLIENT_EXE} label namespace  ${TOPO} istio-injection=enabled --overwrite=true
+
+    if [ "${IS_OPENSHIFT}" == "true" ]; then
+      apply_network_attachment ${TOPO}
+    fi
+
+    ${CLIENT_EXE} apply -n ${TOPO}  -f ${BASE_URL}/topology-generator/deploy/generator.yaml
+}
+
+generate_config() {
+
+  if [ "${IS_OPENSHIFT}" == "true" ]; then
+    for (( i=1; i<=$NUM_NS; i++ ))
+        do
+
+            apply_network_attachment n${i}
+    done
+  fi
+}
+
+while [ $# -gt 0 ]; do
+  key="$1"
+  case $key in
+    -c|--client)
+      CLIENT_EXE="$2"
+      shift;shift
+      ;;
+    -n|--num_ns)
+      NUM_NS="$2"
+      shift;shift
+      ;;
+    -g|--generate_config)
+      GENERATE_CONFIG="$2"
+      shift;shift
+      ;;
+    -d|--delete)
+      DELETE_DEMOS="$2"
+      shift;shift
+      ;;
+    -h|--help)
+      cat <<HELPMSG
+Valid command line arguments:
+  -c|--client: either 'oc' or 'kubectl'
+  -n|--num_ns: number of namespaces to be generate in the generator to apply the network policies
+  -d|--delete: if 'true' demos will be deleted; otherwise, they will be installed
+  -g|--generate_config: Generate configuration of the topology based in the number of namespaces (-n)
+  -h|--help: this text
+
+Usage:
+  1) Run with no args - or -c - to install the topology generator
+  2) Run the proxy: CLIENT port-forward svc/topology-generator 8080:8080 -n topology-generator
+  3) Go to the app and generate the topology
+  4) Run -g to run the openshift configuration and -n with the number of namespaces created
```

Comment:
I was confused here and had to look at the code to see what this is doing. It isn't clear what this "generate-config" option does. I think we should make the help text more clear.

IIUC, this option is only for OpenShift AND upstream Istio - if you are on Minikube or other non-OpenShift cluster, you do not need this. If you are using OSSM, you do not need this. But if you are on OpenShift using upstream Istio, this `generate-config` will create the necessary NAD and SCCs that are required.

So perhaps provide a little more detail here to help the user understand what this option is and when it is needed. I had to read the code to figure it out.

Something like:

`4) Run -g to configure the demo to work on OpenShift. You only need to run this if you are on OpenShift and using upstream Istio (not OSSM).`

Reply:
Updated help to clarify this behavior